skimage/filters/_gaussian.py

```diff
@@ -113,8 +117,13 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
     image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output is None:
+        output = np.empty_like(image)
+    elif not np.issubdtype(output.dtype, np.floating):
+        raise ValueError("Provided output data type is not float")
+    ndi.gaussian_filter(image, sigma, output=output, mode=mode, cval=cval,
```

Comment:
`output = ` ??? This would allow scipy to allocate things right?

Reply:
The problem is that in scipy version 0.19, which is still supported by skimage, if `output` is not `None`, `ndi.gaussian_filter` returns None :-(

Reply:
I see. 

Yeah. Let's leave this PR as is and maybe bump up scipy minimum version.

Reply:
@hmaarrfk different versions of `scipy.ndimage.gaussian_filter` has different behavior, when output is None.